### ISSUE TITLE
proxy3270 crashes sometimes

### ISSUE THREAD

**Issue by moshix** [racingmars/proxy3270#6]
like in this case:

goroutine 14 [running]:
github.com/racingmars/go3270.decodeBufAddr(...)
        /home/moshe/go/pkg/mod/github.com/racingmars/go3270@v0.0.0-20201226094253-eddc39723a84/response.go:213
github.com/racingmars/go3270.readPosition(0x5c26c0, 0xc0000b8018, 0x2f1, 0x0, 0x0, 0x0, 0x0)
        /home/moshe/go/pkg/mod/github.com/racingmars/go3270@v0.0.0-20201226094253-eddc39723a84/response.go:130 +0x28e
github.com/racingmars/go3270.readResponse(0x5c26c0, 0xc0000b8018, 0xc000103790, 0x36e, 0x226, 0x0, 0x0, 0xc000024b00, 0x2)
        /home/moshe/go/pkg/mod/github.com/racingmars/go3270@v0.0.0-20201226094253-eddc39723a84/response.go:80 +0xaa
github.com/racingmars/go3270.ShowScreen(0xc0000a3c00, 0x15, 0x20, 0xc0001580c0, 0x2, 0x21, 0x5c26c0, 0xc0000b8018, 0x7d, 0x21, ...)
        /home/moshe/go/pkg/mod/github.com/racingmars/go3270@v0.0.0-20201226094253-eddc39723a84/screen.go:150 +0x3c6
github.com/racingmars/go3270.HandleScreen(0xc0000a3c00, 0x15, 0x20, 0xc00009a7b0, 0x0, 0xc000103f2f, 0x1, 0x1, 0xc000103f2e, 0x1, ..
        /home/moshe/go/pkg/mod/github.com/racingmars/go3270@v0.0.0-20201226094253-eddc39723a84/looper.go:130 +0x631
main.handle(0x5c26c0, 0xc0000b8018, 0x1)
        /home/moshe/proxy3270/main.go:117 +0x299
created by main.main.func1
        /home/moshe/proxy3270/main.go:99 +0xab

**Comment by racingmars** [comment]
Any more details on this? I'm not seeing the error itself in the stack trace snippet above...what was the actual panic? Array out of bounds or something?

**Comment by moshix** [comment]
Oops, sorry wrong cutting and pasting:

the error was: 

panic: runtime error: index out of range [255] with length 255

I guess it's related to larger than Mod2 term sizes...

thanks a lot !

**Comment by racingmars** [comment]
I haven't been able to figure out how to reproduce this. I've added a log print statement where I think this is happening to give some additional details, though. Can you update to the latest checked-in code, and if it happens again, grab the log line that says "UNEXPECTED VALUE: decodeBufAddr got raw value of..." which should get printed out if this panic is about to happen?

**Comment by moshix** [comment]
Just happened again. I noticed it only happens when Vista3270 is connected to proxy3270

panic: runtime error: index out of range [255] with length 255

goroutine 50 [running]:
github.com/racingmars/go3270.decodeBufAddr(0xc00021f1ff, 0xc00019e031)
        /home/moshe/go/pkg/mod/github.com/racingmars/go3270@v0.0.0-20201227093120-f01cf72d6de3/response.go:215 +0x21e
github.com/racingmars/go3270.readPosition(0x5c27c0, 0xc000214010, 0x2f1, 0x0, 0x0, 0x0, 0x0)
        /home/moshe/go/pkg/mod/github.com/racingmars/go3270@v0.0.0-20201227093120-f01cf72d6de3/response.go:132 +0x125
github.com/racingmars/go3270.readResponse(0x5c27c0, 0xc000214010, 0xc00021f790, 0x382, 0x235, 0x0, 0x0, 0xc0001945c0, 0x2)
        /home/moshe/go/pkg/mod/github.com/racingmars/go3270@v0.0.0-20201227093120-f01cf72d6de3/response.go:82 +0xaa
github.com/racingmars/go3270.ShowScreen(0xc000111c00, 0x17, 0x20, 0xc00021fa20, 0x2, 0x21, 0x5c27c0, 0xc000214010, 0x7f6cc5f96e60, 0x7f6cc5f96e60, ...)
        /home/moshe/go/pkg/mod/github.com/racingmars/go3270@v0.0.0-20201227093120-f01cf72d6de3/screen.go:150 +0x3c6
github.com/racingmars/go3270.HandleScreen(0xc000111c00, 0x17, 0x20, 0xc0001087b0, 0x0, 0xc00021ff2f, 0x1, 0x1, 0xc00021ff2e, 0x1, ...)
        /home/moshe/go/pkg/mod/github.com/racingmars/go3270@v0.0.0-20201227093120-f01cf72d6de3/looper.go:130 +0x631
main.handle(0x5c27c0, 0xc000214010, 0x1)
        /home/moshe/proxy3270/main.go:130 +0x299
created by main.main.func1
        /home/moshe/proxy3270/main.go:112 +0x70

**Comment by racingmars** [comment]
When this happens, is Vista3270 showing a terminal size other than 80x24? Or are you setting different sizes in Vista3270 but it's still staying in 80x24 per issue #5?

**Comment by racingmars** [comment]
aha... not sure if it's the same problem or not, because I'm not actually getting a panic, but I'm starting to be able to trigger my new "unexpected value" log line when decoding buffer addresses received from the client. Very suspicious, so I'll see if I can figure out what's going on there and I suspect it's related to this crash. I can also trigger the behavior from c3270 so I don't think it's specifically related to Vista3270. Hopefully it's all related.

...update: I just randomly got the crash. No idea how, but at least it's not just your system! Gives me something to experiment with.

**Comment by racingmars** [comment]
I believe I have identified the root cause of this problem. I've filed a corresponding issue over with go3270: https://github.com/racingmars/go3270/issues/4

During investigation of this problem, I think I've also found another problem (which is what my unexpected values in buffer address decoding was catching) which I still need to investigate.

I'll keep this open here until I can fix go3270 and incorporate the fix in this project.
